test/AST-Quake/if.cpp

```diff
@@ -65,3 +65,63 @@ struct kernel_else {
 // CHECK:             quake.x {{\[}}%[[VAL_7]]] %[[VAL_8]] : (!quake.ref, !quake.ref) -> ()
 // CHECK:           }
 // CHECK:           return %[[VAL_1]] : i32
+
+struct kernel_short_circuit_and {
+  __qpu__ int operator()() {
+    cudaq::qreg reg(3);
+    if (mz(reg[0]) && mz(reg[1]))
+      x(reg[2]);
+    return 0;
+  }
+};
+
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__kernel_short_circuit_and() -> i32 attributes {"cudaq-entrypoint", "cudaq-kernel"} {
+// CHECK:           %[[VAL_0:.*]] = arith.constant false
+// CHECK:           %[[VAL_1:.*]] = arith.constant 0 : i32
+// CHECK:           %[[VAL_2:.*]] = quake.alloca !quake.veq<3>
+// CHECK:           %[[VAL_3:.*]] = quake.extract_ref %[[VAL_2]][0] : (!quake.veq<3>) -> !quake.ref
+// CHECK:           %[[VAL_4:.*]] = quake.mz %[[VAL_3]] : (!quake.ref) -> i1
+// CHECK:           %[[VAL_5:.*]] = arith.cmpi eq, %[[VAL_4]], %[[VAL_0]] : i1
```

Comment:
The `bool` comparison seemed like it might be avoided, since we're dealing with `bool`s in the first place. However, I can see where it makes both the `&&` and `||` cases follow a similar structure as well.